lib/bowling_score.rb

```diff
@@ -0,0 +1,44 @@
+class BowlingGame
+  attr_reader :frames, :in_turn, :total_score
+
+  def initialize
+    @frames = Array.new
+    @in_turn = false
+    @bowls = Array.new
+    @total_score = 0
+  end
+
+  def roll(pins_downed)
+    check_turn
+    turn(pins_downed)
+  end
```

Comment:
I think this is what I should have done, rather than processing frames, which made calculating scores for consecutive strikes a massive headache